app/src/main/java/de/danoeh/antennapod/dialog/IntraFeedSortDialog.java

```diff
@@ -16,35 +16,59 @@
     @NonNull
     protected Context context;
 
-    public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder) {
+    final private String[] sortItems;
+    final private SortOrder[] sortValues;
+
+    public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder, @NonNull boolean isLocalFeed) {
         this.context = context;
         this.currentSortOrder = sortOrder;
+
+        if (isLocalFeed) {
+            sortItems = context.getResources().getStringArray(R.array.local_feed_episodes_sort_options);
+            final String[] localSortStringValues =
+                    context.getResources().getStringArray(R.array.local_feed_episodes_sort_values);
+            sortValues = SortOrder.valuesOf(localSortStringValues);
+        } else {
+            sortItems = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
+            final String[] commonSortStringValues =
+                    context.getResources().getStringArray(R.array.feed_episodes_sort_values);
+            sortValues = SortOrder.valuesOf(commonSortStringValues);
+        }
     }
 
     public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
-        final String[] valueStrs = context.getResources().getStringArray(R.array.feed_episodes_sort_values);
-        final SortOrder[] values = new SortOrder[valueStrs.length];
-        for (int i = 0; i < valueStrs.length; i++) {
-            values[i] = SortOrder.valueOf(valueStrs[i]);
-        }
+        final SortOrder[] values = sortValues;
+        final String [] items = sortItems;
+
+        int idxCurrentSort = getCurrentSortOrderIndex(values);
 
-        int idxCurrentSort = 0;
-        for  (int i = 0; i < values.length; i++) {
+        AlertDialog.Builder builder =
+                new AlertDialog.Builder(context)
+                        .setTitle(R.string.sort)
+                        .setSingleChoiceItems(items, idxCurrentSort, (dialog, idxNewSort) -> {
+                            updateSort(values[idxNewSort]);
+                            dialog.dismiss();
+                        })
+                        .setNegativeButton(R.string.cancel_label, null);
+        builder.create().show();
+    }
+
+    /**
+     * Retrieves index of currentSortOrder index in values array.
+     * @param values - array of available sorting options
+     * @return if currentSortOrder is found in array - returns index of that element,
+     *         otherwise returns -1;
+     */
+    private int getCurrentSortOrderIndex(SortOrder[] values) {
```

Comment:
Can't you just use the member variable directly instead of passing a copy here? Currently, the method uses `values` as parameter but `currentSortOrder` as a member variable, which is a bit confusing when reading the method.

Reply:
Yes, definitely better to do that, should have noticed this too as part of the refactor.